Source/Scene/PointCloud3DTileContent.js

```diff
@@ -0,0 +1,1267 @@
+    }
+
+    var positionLocation = 0;
+    var colorLocation = 1;
+    var normalLocation = 2;
+    var batchIdLocation = 3;
+    var numberOfAttributes = 4;
+
+    function createResources(content, frameState) {
+        var context = frameState.context;
+        var parsedContent = content._parsedContent;
+        var pointsLength = content._pointsLength;
+        var positions = parsedContent.positions;
+        var colors = parsedContent.colors;
+        var normals = parsedContent.normals;
+        var batchIds = parsedContent.batchIds;
+        var styleableProperties = parsedContent.styleableProperties;
+        var hasStyleableProperties = defined(styleableProperties);
+        var isQuantized = content._isQuantized;
+        var isOctEncoded16P = content._isOctEncoded16P;
+        var isRGB565 = content._isRGB565;
+        var isTranslucent = content._isTranslucent;
+        var hasColors = content._hasColors;
+        var hasNormals = content._hasNormals;
+        var hasBatchIds = content._hasBatchIds;
+
+        var batchTable = content._batchTable;
+        var hasBatchTable = defined(batchTable);
+
+        var styleableVertexAttributes = [];
+        var styleableShaderAttributes = {};
+        content._styleableShaderAttributes = styleableShaderAttributes;
+
+        if (hasStyleableProperties) {
+            var attributeLocation = numberOfAttributes;
+
+            for (var name in styleableProperties) {
+                if (styleableProperties.hasOwnProperty(name)) {
+                    var property = styleableProperties[name];
+                    var typedArray = property.typedArray;
+                    var componentCount = property.componentCount;
+                    var componentDatatype = ComponentDatatype.fromTypedArray(typedArray);
+
+                    var vertexBuffer = Buffer.createVertexBuffer({
+                        context : context,
+                        typedArray : property.typedArray,
+                        usage : BufferUsage.STATIC_DRAW
+                    });
+
+                    content._vertexMemorySizeInBytes += vertexBuffer.sizeInBytes;
+
+                    var vertexAttribute = {
+                        index : attributeLocation,
+                        vertexBuffer : vertexBuffer,
+                        componentsPerAttribute : componentCount,
+                        componentDatatype : componentDatatype,
+                        normalize : false,
+                        offsetInBytes : 0,
+                        strideInBytes : 0
+                    };
+
+                    styleableVertexAttributes.push(vertexAttribute);
+                    styleableShaderAttributes[name] = {
+                        location : attributeLocation,
+                        componentCount : componentCount
+                    };
+                    ++attributeLocation;
+                }
+            }
+        }
+
+        var uniformMap = {
+            u_pointSize : function() {
```

Comment:
Maybe this can wait until the push towards WebGL2 UBOs, I can think of a few area throughout 3D Tiles where this comment applies.